src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -944,3 +944,59 @@ void C2_MacroAssembler::neon_compare(FloatRegister dst, BasicType bt, FloatRegis
     }
   }
 }
+
+void C2_MacroAssembler::sve_compare(PRegister pd, BasicType bt, PRegister pg,
+                                    FloatRegister zn, FloatRegister zm, int cond) {
+  SIMD_RegVariant size = elemType_to_regVariant(bt);
+  if (bt == T_FLOAT || bt == T_DOUBLE) {
+    switch (cond) {
+      case BoolTest::eq: sve_fcmeq(pd, size, pg, zn, zm); break;
```

Comment:
Wouldn't this work better by making cond and size arguments to sve_fcm ?

Reply:
Thanks! But we still need to map c2 specific condition value in BoolTest to instruction condition encoding. So, I think making cond arguments to fcm doesn't make help here. (This also aligns current NEON implementation.) What do you think?

Reply:
> we still need to map c2 specific condition value in BoolTest to instruction condition encoding

You'd still need a way to do that mapping, but it'd be a simple switch in a function that does the conversion to Assembler::Condition. Then you'd just have e.g. sve_cmp(GE, xword ...) and most of this code would disappear. You wouldn't need all the Assembler functions, either.

Reply:
> 
> You'd still need a way to do that mapping, but it'd be a simple switch in a function that does the conversion to Assembler::Condition. Then you'd just have e.g. sve_cmp(GE, xword ...) and most of this code would disappear. You wouldn't need all the Assembler functions, either.

But Assembler::Condition cannot be encoded to vector compare instructions directly, which means that we need two switch mappings: BoolTest --> Assembler::Condition --> vector compare encodings. That looks more complicated than current implementation?

Reply:
Ah, true, SVE conditions are (of course!) different from scalar conditions. But that's still no more than a mapping function from c2-specific cond to SVE codes, isn't it? But I take your point, it's not so obvious as I thought.
(Re the "aligning with Neon" argument: I let a lot of things through in the past, which I now regret. I should have pushed back harder.)

Reply:
> Ah, true, SVE conditions are (of course!) different from scalar conditions. But that's still no more than a mapping function from c2-specific cond to SVE codes, isn't it? 

Updated in the latest commit.